### PR TITLE
Turpentine and Camphine

### PR DESCRIPTION
**Pull request by Sathra225**
#### Summary
Content "Turpentine and Camphine"

#### Purpose of change
With the change to lamp oil/kerosene, getting fuel for oil lamps is only scavenging or a rare appliance. Adds turpentine and camphine (the interim oil lamp fuel between whale oil and kerosene) plus crafting recipes for both.

#### Describe the solution
Adds turpentine and camphine (turpentine/ethanol blend used for oil lamps). Also adds pine resin and harvesting it from pine trees.

#### Describe alternatives you've considered
Making this Innawoods only, but I could already guess I'd be asked for this to be in mainline so...
Having clay oil lamps be able to use cooking oils instead of kerosene, but couldn't get it to work.
Some other way to get pine resin, but couldn't figure out a way to do it within my capabilities.

#### Testing
Made all the changes, harvested pine resin, cooked it up, made camphine, filled oil lamp with it and lit lamp. Nothing exploded.

#### Additional context
Chemical data for turpentine from here: https://iopscience.iop.org/article/10.1088/1757-899X/811/1/012015/pdf and https://www.engineeringtoolbox.com/fuels-higher-calorific-values-d_169.html
I couldn't find some of the data in the right format (like fuel energy) but apparently its roughly equivalent to diesel so went with that.
Production is from https://www.youtube.com/watch?v=t7zT9zOpBA4&t=683s and the other videos on the channel, plus various  other sources that agreed with it. The specific details for the turpentine were that non-fresh sap (scrape) was about half as efficient as tapped (gum). Drip was roughly 6 barrels to 1 gum to turpentine. 2 parts gum to 1 part water, plus water for cooling.

### CODE REVIEW

**Comment by Sathra225** [comment]
Draft while I work on spawns of turpentine and camphine.

**Comment by Sathra225** [comment]
Basic spawn locations added and tested. Ready for review.

File: data/json/recipes/recipe_medsandchemicals.json [hunk 1]
```diff
@@ -1658,5 +1658,50 @@
     "tools": [ [ [ "syringe", -1 ] ], [ [ "surface_heat", 30, "LIST" ] ] ],
     "components": [ [ [ "venom_paralytic", 1 ] ] ],
     "book_learn": [ [ "textbook_toxicology", 5 ], [ "textbook_botany", 5 ] ]
+  },
+  {
+    "type": "recipe",
+    "activity_level": "LIGHT_EXERCISE",
+    "result": "chem_turpentine",
+    "byproducts": [ [ "rosin", 1 ] ],
```
Comment (Night-Pryanik) [inline data/json/recipes/recipe_medsandchemicals.json:1666]: Just want to be sure that it's not a typo. Did you mean resin?

Comment (Sathra225) [reply data/json/recipes/recipe_medsandchemicals.json:1666]: Its meant to be rosin. Turpentine is the volatile oils, rosin is the goop left behind. The seive is due to how there'll be a bunch of splinters and crap with how the pine resin is harvested which need to be seperated before allowing the rosin to harden.

**Comment by Maleclypse** [comment]
Can you add camphine to the dictionary file so it doesn't cause every PR after this to fail the spellchecker?

**Comment by Sathra225** [comment]
> Can you add camphine to the dictionary file so it doesn't cause every PR after this to fail the spellchecker?

done

**Comment by Light-Wave** [comment]
Yeah, this will work. Thank you for your work on keeping Innawood playable. It's much appreciated.

One could consider if recipes where pine bough is used as glue should be changed, but that can be done later.

As for getting cooking oil to work, perhaps it could work to first craft it into another item, like mixed oils, much like how we have mixed alcohol? Then again, perhaps that type of fuel generates too much smoke and too little light IRL for mainline to want it any more.

**Comment by Sathra225** [comment]
Yeah, the plan was to have the clay oil lamp use cooking oil as it is, as far as I can tell, roughly like a liquid-fuelled candle, with the glass and metal one using 'proper' fuel. Then boost the light level of the better oil lamp to 3x3.
I've made oil-float candles (oil floating on water with a disk that floats on the oil that the wick sticks out of) using generic vegetable oil. It's not much more smokey than a wax candle, and is pretty much just a candle you can refuel.

**Review by Light-Wave — APPROVED** [review]

**Pull request closed without merging**